neqo-transport/src/send_stream.rs

```diff
@@ -520,7 +520,7 @@ impl SendStream {
         match self.state {
             SendStreamState::Send { ref mut send_buf } => {
                 send_buf.mark_as_acked(offset, len);
-                if send_buf.buffered() < TX_STREAM_BUFFER {
+                if self.avail() > 0 && self.flow_mgr.borrow().conn_credit_avail() > 0 {
```

Comment:
Could you instead move the connection flow control check to self.avail() ?  Running min on that seems like it might solve some problems.